sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1439,6 +1458,12 @@ pi_result _pi_context::getAvailableCommandList(
   // Immediate commandlists have been pre-allocated and are always available.
   if (Queue->Device->useImmediateCommandLists()) {
     CommandList = Queue->getQueueGroup(UseCopyEngine).getImmCmdList();
+    if (CommandList->second.EventList.size() >
+        ImmCmdListsEventCleanupThreshold) {
+      std::vector<pi_event> EventListToCleanup;
+      Queue->resetCommandList(CommandList, false, EventListToCleanup);
+      CleanupEventListFromResetCmdList(EventListToCleanup, true);
+    }
```

Comment:
Because currently for immediate command lists we cleanup only at point of synchronization.  If application doesn't have any points of synchronization then we keep creating new events for immediate  lists. This change results in better recycling of events and I see significant performance improvement for on of the sycl benchmarks.